### ISSUE TITLE
Change action "links" to buttons

### ISSUE THREAD

**Issue by robincornett** [easydigitaldownloads/easy-digital-downloads#7705]
### Explain your enhancement (please be detailed)
In places where EDD is triggering JavaScript to change something on the page, `<button>` should be used instead of an empty `<a>` to be semantically correct and improve accessibility.
* [Links, Buttons, Submits, and Divs...](https://adrianroselli.com/2016/01/links-buttons-submits-and-divs-oh-hell.html#Button)
* [Links and Hypertext (specifically the "Bad Example")](https://webaim.org/techniques/hypertext/)

I'm looking specifically at the actions for repeatable rows (examples: [Remove](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/master/includes/admin/downloads/metabox.php#L443); [Show advanced settings](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/master/includes/admin/downloads/metabox.php#L440))

### Justification or use case
This will make the action markup semantically correct and more accessible. It appears that the JS is triggered off off the class alone, so changing the element type should be sufficient to fix. Additional CSS may be needed as well (adding a `button-secondary` class inherits default WordPress button styling, but also changes the height of the title row).

It looks like this was raised at one point in #4343, which has been closed.

**Comment by cklosowski** [comment]
I’m setting this for a 3.1 milestone. Right now 3.0 is to close to a massive rewrite of all the JS and because we’re changing the markup that some people might be looking for clicks on, this has to go in a major.

**Comment by robincornett** [comment]
FWIW, the original issue was closed with the final comment being that the PR was being merged into 2.6. I did some digging and it looks like the faux links were changed to buttons, but then reverted (I suspect unintentionally) back to link markup as part of the work done on #2704 (there was a lot).

I did some testing and correcting this issue without changing CSS classes does not impact the JS working in 2.9 or 3.0.

**Comment by spencerfinnell** [comment]
@robincornett The one thing we can't be certain about is people doing something like:

```js
$( 'a.remove' ).on( 'click', function() { } );
```

It's unlikely, and subjectively not a great practice for this very reason, but is possible.

**Comment by cklosowski** [comment]
Moving outside of EDD 3.1 as this will take a large effort to test with 3rd party compatibility.
